### PR TITLE
m68k analysis fixes:

### PR DESCRIPTION
**Pull request by fabrice-martinez**
- Fill op->jump and op->fail for all branch and jump instructions
- Handle short and word variants of BSR (BSR.S and BSR.W)
- DBxx instructions treated as conditional branches
- Handle PC relative variant of JMP & JSR
- Bugfix for the decoding of long instructions (code in asm_m68k_cs.c assumed a maximum of 8 bytes, but maximum is 10 bytes)

### CODE REVIEW

**Comment by radare** [comment]
Fix indent issues, move defines to the top of the file. And fix build because theres a missing symbol

libr_anal.so: undefined reference to `handle_jump_instruction' collect2: error: ld returned 1 exit status make[2]: *** [rax2] Error 1 make[1]: *** [all] Error 1 make: *** [all] Error 2 + exit 1

File: libr/anal/p/anal_m68k_cs.c [hunk 1]
```diff
@@ -26,6 +26,41 @@
 #define MEMINDEX(x) insn->detail->m68k.operands[x].mem.index
 #define MEMDISP(x) insn->detail->m68k.operands[x].mem.disp
 
+inline ut64 make_64bits_address(ut64 address)
```
Comment (radare) [inline libr/anal/p/anal_m68k_cs.c]: make this static

Comment (radare) [reply libr/anal/p/anal_m68k_cs.c]: bracess start in the same line

File: libr/anal/p/anal_m68k_cs.c [hunk 2]
```diff
@@ -26,6 +26,41 @@
...
+{
+	return UT32_MAX & address;
+}
+
+inline void handle_branch_instruction(RAnalOp *op, ut64 addr, cs_m68k *m68k, ut32 type, int index)
+{
+	#if CS_API_MAJOR >= 4
```
Comment (radare) [inline libr/anal/p/anal_m68k_cs.c]: #if aligns at column 0

File: libr/anal/p/anal_m68k_cs.c [hunk 3]
```diff
@@ -26,6 +26,41 @@
...
+		if (m68k->operands[index].type == M68K_OP_BR_DISP) {
+			op->type = type;
+			// TODO: disp_size is ignored
+			op->jump = make_64bits_address(addr + m68k->operands[index].br_disp.disp + 2);
```
Comment (radare) [inline libr/anal/p/anal_m68k_cs.c]: missing space before (

File: libr/anal/p/anal_m68k_cs.c [hunk 4]
```diff
@@ -26,6 +26,41 @@
...
+		op->fail = make_64bits_address(addr + op->size);
+#endif
+}
+
+inline void handle_jump_instruction(RAnalOp *op, ut64 addr, cs_m68k *m68k, ut32 type)
+{
+	op->type = type;
+
+	// Handle PC relative mode jump
+	if (m68k->operands[0].address_mode == M68K_AM_PCI_DISP)
```
Comment (radare) [inline libr/anal/p/anal_m68k_cs.c]: add braces {}

File: libr/anal/p/anal_m68k_cs.c [hunk 5]
```diff
@@ -232,28 +267,13 @@ static int analop(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len, RAn
...
-			op->jump = addr + m68k->operands[0].br_disp.disp + 2;
-			op->fail = addr + insn->size;
-		}
-#else
-		op->type = R_ANAL_OP_TYPE_CJMP;
-		// TODO: disp_size is ignored
-		op->jump = addr + m68k->operands[0].br_disp.disp + 2;
-		op->fail = addr + insn->size;
-#endif
+		handle_branch_instruction(op, addr, m68k, R_ANAL_OP_TYPE_CJMP, 0);
```
Comment (radare) [inline libr/anal/p/anal_m68k_cs.c]: missing space before (

File: libr/asm/p/asm_m68k_cs.c [hunk 6]
```diff
@@ -72,9 +72,12 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
...
 		goto beach;
 	}
-	ut8 mybuf[8] = {0};
-	int mylen = R_MIN (8, len);
-	memcpy (mybuf, buf, R_MIN (8, len));
+
+	#define M68K_LONGEST_INSTRUCTION 10
+	ut8 mybuf[M68K_LONGEST_INSTRUCTION] = {0};
+	int mylen = R_MIN (M68K_LONGEST_INSTRUCTION, len);
+	memcpy (mybuf, buf, R_MIN (M68K_LONGEST_INSTRUCTION, len));
```
Comment (radare) [inline libr/asm/p/asm_m68k_cs.c]: reuse mylen instead of recomputing twice

**Comment by radare** [comment]
Also, most compilers ignore the inline directive

**Comment by radare** [comment]
can you also submit another PR implementing ESIL support for m68k? that will improve a lot the disassembly quality and open the door for emulating portions of code.

**Comment by fabrice-martinez** [comment]
I can look into that. Look complex if we want to emulate supervisor / user modes, exceptions, undocumented flags, ...

**Comment by radare** [comment]
Yeah, but for now just focus on the general purpose instructions which are the most common.

**Pull request merged**